crates/core/src/db/datastore/locking_tx_datastore/mod.rs

```diff
@@ -1673,6 +1675,37 @@ impl Locking {
         }
         Ok(())
     }
+
+    pub fn module_hash(&self, tx: &MutTxId) -> Result<Option<Hash>, DBError> {
+        match tx.lock.iter(&ST_MODULE_ID)?.next() {
+            None => Ok(None),
+            Some(data) => {
+                let row = StModuleRow::try_from(data.view())?;
+                Ok(Some(row.program_hash))
+            }
+        }
+    }
+
+    pub(crate) fn set_module_hash(&self, tx: &mut MutTxId, token: u64, hash: Hash) -> Result<(), DBError> {
```

Comment:
These should probably not be inherents, but I couldn't make up my mind yet how to make them fit into the trait structure.